packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -479,262 +1013,250 @@ const DialogueCheckBox: React.FC<{
+const DialogueNextStepsButton: React.FC<DialogueNextStepsButtonProps> = ({
   isMatched,
+  checkId,
   targetUserId,
   targetUserDisplayName,
   currentUser,
-  loadingNewDialogue,
-  createDialogue,
   classes,
 }) => {
-  if (!isMatched) return <div></div>; // need this instead of null to keep the table columns aligned
 
-  return (
-    <div>
-      <button
-        className={classes.newDialogueButton}
-        onClick={(e) =>
-          createDialogue(
-            `${currentUser?.displayName}/${targetUserDisplayName}`,
-            [targetUserId]
-          )
-        }
-      >
-        {loadingNewDialogue ? <a data-cy="message">Creating New Dialogue...</a> : <a data-cy="message">Start Dialogue</a>}
+  const { openDialog } = useDialog();
+
+  const navigate = useNavigate();
+
+  const {loading: userLoading, results} = useMulti({
+    terms: {
+      view: "dialogueMatchPreferences",
+      dialogueCheckId: checkId,
+      limit: 1000,
+    },
+    fragmentName: "DialogueMatchPreferenceInfo",
+    collectionName: "DialogueMatchPreferences",
+  });
+
+  const { document: dialogueCheck } = useSingle({
+    fragmentName: "DialogueCheckInfo",
+    collectionName: "DialogueChecks",
+    documentId: checkId,
+  });
+
+  if (!isMatched) return <div className={classes.hideAtXs}></div>; // need this instead of null to keep the table columns aligned
+
+  const userMatchPreferences = results?.[0]
+  const generatedDialogueId = userMatchPreferences?.generatedDialogueId;
+
+  if (!!generatedDialogueId) {
+    return (
+      <button className={classes.lightGreenButton} onClick={(e) => redirect(generatedDialogueId, navigate)}>
+        <a data-cy="message">Go to dialogue</a>
       </button>
-    </div>
+    );
+  }
+
+  if (userMatchPreferences) {
+    return (
+      <div className={classes.waitingMessage}>
+        Waiting for {targetUserDisplayName}...
+      </div>
+    );
+  }
+
+  return (
+    <button
+      className={classNames(classes.enterTopicsButton, isMatched ? '' : classes.hideAtXs)}
+      onClick={(e) => {
+        dialogueCheck && openDialog({
+          componentName: 'NextStepsDialog',
+          componentProps: {
+            userId: currentUser?._id,
+            targetUserId,
+            targetUserDisplayName,
+            dialogueCheckId: checkId,
+            dialogueCheck
+          }
+        })
+      }}
+    >
+      <a data-cy="message">Enter topics</a>
+    </button>
   );
 };
 
 const MessageButton: React.FC<{
   targetUserId: string;
-  currentUser: UsersCurrent; 
-  classes: ClassesType;
-}> = ({ targetUserId, currentUser, classes }) => {
+  currentUser: UsersCurrent;
+  isMatched: boolean;
+  classes: ClassesType<typeof styles>;
+}> = ({ targetUserId, currentUser, isMatched, classes }) => {
   const { NewConversationButton } = Components;
   
   return (
-    <button className={classes.messageButton}>
+    <button className={classNames(classes.messageButton, isMatched ? classes.hideAtXs : '')}>
```

Comment:
Conditional classNames like `classes.hideAtXs` should be implemented like so:
```
classNames(classes.messageButton, {
  [classes.hideAtXs]: isMatched
})
```

Reply:
@rjmk